### PR TITLE
✨ amp-video-iframe: pass consent data to iframe

### LINKED ISSUE #31826: amp-video-iframe should pass consent
Probably by using the [same method as `amp-iframe`.](https://amp.dev/documentation/components/amp-iframe/#iframe-&-consent-data)

### PR DESCRIPTION
**Pull request by alanorozco**
Fixes #31826

### CODE REVIEW

File: extensions/amp-video-iframe/amp-video-iframe.md
Comment (alanorozco) [inline extensions/amp-video-iframe/amp-video-iframe.md:448]: @micajuine-ho It would be great to describe what these keys are, but I don't have that information. Do we document `consentMetadata`, `consentString` and `consentPolicyState` anywhere?

File: src/consent.js
Comment (alanorozco) [inline src/consent.js:126]: I extracted what `amp-iframe` does and refactored it somewhat.

I feel like there's a better name for this function, but I'm not sure what to call the returned object other than `consents`. Suggestions welcome.

File: examples/amp-video-iframe/consent.html
Comment (micajuine-ho) [inline examples/amp-video-iframe/consent.html:41]: I think in most cases, this element would also be blocked by `amp-consent` via the `data-block-on-consent` attribute.

File: examples/amp-video-iframe/consent.html
Comment (micajuine-ho) [inline examples/amp-video-iframe/consent.html:49]: nit: removed `consents`, it's deprecated.

File: src/consent.js
Comment (micajuine-ho) [reply src/consent.js:126]: IMO `collectConsents` makes me think about actually prompting some consent UI. I think something more specific like `getCollectedConsentData` is more clear.

File: src/consent.js
Comment (micajuine-ho) [inline src/consent.js:107]: nit: `policyId = 'default'`

**Review by micajuine-ho — COMMENTED** [review]
In general I am good with the additional helper method in `src/consent`, however I do want to know if 

1.) This method will be used by any other extensions in future and
2.) If we will need to add any other consent data (i.e. shared data)

In the past, we had stayed away from packaging all the data into a single helper method, since it was up to the vendor to decide what information they needed from AMP/CMPs--i.e. server side decisions following TCF IAB 2.0 need the consent string to parse and interpret, whereas other vendors just need the consent state. 

However, it does seem like it's prudent in this case since amp-iframe and amp-video-iframe are similar and assume that vendors need this exact date in this format.

File: extensions/amp-video-iframe/amp-video-iframe.md
Comment (micajuine-ho) [reply extensions/amp-video-iframe/amp-video-iframe.md:448]: We document them here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/customizing-extension-behaviors-on-consent.md#advanced-blocking-behaviors

File: src/consent.js
Comment (alanorozco) [reply src/consent.js:107]: In this case the argument could be both`null` or `undefined` so default params won't work (they're only set for `undefined`).

image

File: examples/amp-video-iframe/consent.html
Comment (alanorozco) [reply examples/amp-video-iframe/consent.html:41]: I added a note to component docs:

https://github.com/ampproject/amphtml/blob/564ad74d934e2afd85df683fe45a3ac947318f8b/extensions/amp-video-iframe/amp-video-iframe.md#L432

Would that be sufficient?

**Comment by alanorozco** [comment]
> In the past, we had stayed away from packaging all the data into a single helper method, since it was up to the vendor to decide what information they needed from AMP/CMPs--i.e. server side decisions following TCF IAB 2.0 need the consent string to parse and interpret, whereas other vendors just need the consent state.

Is it expensive to get any of these values? If not, maybe we should just send all of them in some cases.

File: src/consent.js
Comment (micajuine-ho) [reply src/consent.js:107]: Interesting! Thanks for the tidbit

File: examples/amp-video-iframe/consent.html
Comment (micajuine-ho) [reply examples/amp-video-iframe/consent.html]: Sorry, what I meant was that if the component is not blocked with `data-block-on-consent`, then `getConsentData` will not return the collect consent information. It needs to be blocked w/ `data-block-on-consent` (it will unblock immediately, if we have stored consent data)

**Comment by micajuine-ho** [comment]
> Is it expensive to get any of these values? If not, maybe we should just send all of them in some cases.

Not much more expensive but shared data could be very large--we don't have a limit on the amount of data passed in it.

**Comment by alanorozco** [comment]
As @micajuine-ho and I discussed offline: The size of `sharedData` is expected to be up to 1K, which shouldn't be an issue for callers (they usually set an iframe's name, or pass it by `postMessage`, for which 1K doesn't cause a problem)

File: src/consent.js
Comment (alanorozco) [reply src/consent.js]: Renamed to `getConsentDataToForward`.

File: examples/amp-video-iframe/consent.html
Comment (micajuine-ho) [inline examples/amp-video-iframe/consent.html:51]: Related to https://github.com/ampproject/amphtml/issues/31568

Would love to talk more about how we can create some default or pub generated UI for elements blocked by rejected consent.

File: extensions/amp-consent/customizing-extension-behaviors-on-consent.md
Comment (micajuine-ho) [inline extensions/amp-consent/customizing-extension-behaviors-on-consent.md:58]: Can you also include amp-iframe in here?

**Review by micajuine-ho — APPROVED** [review]
LGTM!

**Comment by carlosplayinc** [comment]
Hi @alanorozco , thanks for adding this new feature in. I'm trying to integrate amp-consent into amp-video-iframe, but I'm getting an error, even after following the docs. By the looks of this commit, it doesn't seem like your changes have gone out? So the new getConsentData method is not available yet. Are there any plans to fix so that this new method will be available live?

**Comment by alanorozco** [comment]
@carlosplayinc This change wasn't live when you commented, **it should be live now**. Give it another try and please file an issue if necessary.

**Pull request merged**
